### PR TITLE
Replace print by the Python logging framework (RFC)

### PR DESCRIPTION
**Pull request by vdbergh**
Log messages with level >= ERROR give an event log message.

Also: move scheduler to a separate file scheduler.py.

### CODE REVIEW

File: server/fishtest/rundb.py [hunk 1]
```diff
@@ -723,13 +725,12 @@ def scavenge_dead_tasks(self):
...
             task = run["tasks"][task_id]
-            print(
+            self.logger.info(
                 "dead task: run: https://tests.stockfishchess.org/tests/view/{} task_id: {} worker: {}".format(
                     run["_id"], task_id, worker_name(task["worker_info"])
                 ),
-                flush=True,
             )
-            self.handle_crash_or_time(run, task_id)
+            self.handle_clrash_or_time(run, task_id)
```
Comment (vondele) [inline server/fishtest/rundb.py:733]: typo

**Comment by ppigazzini** [comment]
If I'm not wrong you are using the root logger, the best practice is to log at the module level, see at example
<https://betterstack.com/community/guides/logging/python/python-logging-best-practices/>

```python
class ActionDb:
    def __init__(self, db):
        self.db = db
        self.actions = self.db["actions"]
        self.logger = logging.getLogger(__name__)
```

**Comment by vdbergh** [comment]
I am not using the root logger but a child logger with name "fishtest". I am not using `__name__` since I am using the same logger everwhere.

I had quite a bit of trouble with this PR since the tutorials did not work. The reason seems to be that pserve already does some configuration. Switching to a child logger fixed things.

File: server/fishtest/rundb.py (hunk 1 above)
Comment (vdbergh) [reply server/fishtest/rundb.py]: Thanks! Fixed.

**Comment by ppigazzini** [comment]
Pyramid example logs at module level, <https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/logging.html>

The logger configuration in Pyramid is in the usual ini files:

https://github.com/official-stockfish/fishtest/blob/322eea2e377023df3393d577587b9100666949f0/server/production.ini#L37-L67

https://github.com/official-stockfish/fishtest/blob/322eea2e377023df3393d577587b9100666949f0/server/development.ini#L32-L62

**Comment by vdbergh** [comment]
Hmm that's nice! It seems pyramid already does some of the heavy lifting.

It is not clear to me however how one declares a project specific handler. I am using a handler that posts to the event log.

**Comment by vdbergh** [comment]
https://stackoverflow.com/questions/30301824/add-custom-log-handler-to-pyramid

**Comment by vdbergh** [comment]
@ppigazzini The default logging configuration precedes every line with a time stamp. But your logs already seem to have timestamps (provide by systemd?). So I assume you don't want timestamps in the pyramid logs. Is this correct?

**Comment by ppigazzini** [comment]
Systemd already add the time stamp.

**Comment by vdbergh** [comment]
Another problem is that currently the constructor for EventHandler takes rundb.actiondb as argument (which itself is constructed inside rundb.__init__(). It is not obvious to me how handle this without resorting to global variables. 

I guess the EvenHandler will have to open a new connection to the db in the constructor. Not a big deal.

**Comment by ppigazzini** [comment]
If the best practice is to log at module level, `log` is a global variable that somewhat replace the `print` function:

```python
def foo():
    message = "Error!"
    print(message)
```
```python
import logging
log = logging.getLogger(__name__)

def bar():
    message = "Error!"
    log.debug(message)
```

**Comment by vdbergh** [comment]
I want to define a custom logger that also logs to the event log for `level >= ERROR`. 

Unfortunately I don't know how to do this in Pyramid framework (I tried the last two hours or so but there are errors I don't understand).

**Comment by vdbergh** [comment]
I think I am getting to work now.

**Comment by vdbergh** [comment]
Closing in favor of #2067

**Comment by vdbergh** [comment]
This the code for the custom handler...
```python
class EventHandler(logging.Handler):
    def __init__(self):
        super().__init__(level=logging.ERROR)
        logging.disable(level=logging.CRITICAL)
	rundb = RunDb(is_primary_instance=False)
        logging.disable(level=logging.NOTSET)
        self.actiondb = rundb.actiondb

    def emit(self, record):
        message=self.format(record)
	self.actiondb.log_message(
            username="fishtest.system",
            message=message,
        )
```

**Pull request closed without merging**